modules/core/02-client/types/msgs.go

```diff
@@ -283,3 +284,32 @@ func (msg *MsgUpdateParams) ValidateBasic() error {
 	}
 	return msg.Params.Validate()
 }
+
+// NewMsgRecoverClient creates a new MsgRecoverClient instance
+func NewMsgRecoverClient(signer, subjectClientID, substituteClientID string) *MsgRecoverClient {
+	return &MsgRecoverClient{
+		Signer:             signer,
+		SubjectClientId:    subjectClientID,
+		SubstituteClientId: substituteClientID,
+	}
+}
+
+// GetSigners returns the expected signers for a MsgRecoverClient message.
+func (msg *MsgRecoverClient) GetSigners() []sdk.AccAddress {
+	accAddr, err := sdk.AccAddressFromBech32(msg.Signer)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{accAddr}
+}
+
+// ValidateBasic performs basic checks on a MsgRecoverClient.
+func (msg *MsgRecoverClient) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Signer); err != nil {
+		return errorsmod.Wrapf(ibcerrors.ErrInvalidAddress, "string could not be parsed as address: %v", err)
+	}
+	if err := host.ClientIdentifierValidator(msg.SubjectClientId); err != nil {
+		return err
+	}
+	return host.ClientIdentifierValidator(msg.SubstituteClientId)
```

Comment:
nit: adding some breathing room :)  

```suggestion
	if _, err := sdk.AccAddressFromBech32(msg.Signer); err != nil {
		return errorsmod.Wrapf(ibcerrors.ErrInvalidAddress, "string could not be parsed as address: %v", err)
	}
	
	if err := host.ClientIdentifierValidator(msg.SubjectClientId); err != nil {
		return err
	}
	
	return host.ClientIdentifierValidator(msg.SubstituteClientId)
```